### PR TITLE
OJ-2899: Fix address form

### PR DESCRIPTION
**Pull request by MarieseMikely**
### What changed

- Fixed welsh translation bug for house name or number error message 
Screenshot 2024-12-11 at 11 40 07

- Add optional Field Labels for flat number and street name
Screenshot 2024-12-10 at 15 21 19

- Group building address sections as a Fieldset with Legend for house number and house name

- Added aria-describedBy for easier screen reading for accessibility
Screenshot 2024-12-10 at 15 22 08

- Update error structure to be more consistent with `req.error` and 

### Why did it change

To imporove UK address form and to make the international address form consistent update `getIndividualFieldErrorMessages` to be a pure function

### Issue tracking

- [OJ-2899](https://govukverify.atlassian.net/browse/OJ-2899)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2899]: https://govukverify.atlassian.net/browse/OJ-2899?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/address/controllers/address/manual.js [hunk 1]
```diff
@@ -125,6 +138,19 @@ class AddressController extends BaseController {
...
-}
 
+  defaultToFirstField(formFields, first, req) {
+    formFields[first].validate.push({
+      fn: ukBuildingAddressEmptyValidator,
+      message: req.translate("validation.houseNameOrHouseNumber"),
+    });
+  }
+
+  isBuildingAddressEmpty(req) {
```
Comment (Caitlin-cooling) [inline src/app/address/controllers/address/manual.js:149]: This is duplicated from the nonUK controller, right? Do you think we could extract it into a helper?

File: src/app/address/controllers/address/manual.js [hunk 2]
```diff
@@ -29,26 +32,36 @@ class AddressController extends BaseController {
...
         values.checkDetailsHeader = "false";
       }
+
+      addIndividualFieldErrorMessages(
+        "ukBuildingAddressEmptyValidator",
+        req,
+        values
+      );
+
+      values.ukBuildingAddressEmptyValidator = this.isBuildingAddressEmpty(
```
Comment (Caitlin-cooling) [inline src/app/address/controllers/address/manual.js:42]: Please could we call this something like ukBuildingAddressEmptyError? It's not actually a validator is it?

File: src/app/address/controllers/address/manual.js (hunk 1 above)
Comment (Caitlin-cooling) [reply src/app/address/controllers/address/manual.js:149]: Oh the field name is different. Hmmmm these field names are needed for translations, right? So we can't just align them?

File: src/lib/helpers.js [hunk 3]
```diff
@@ -8,4 +8,22 @@ const yearFrom = (year) =>
 const getCountry = (countryCode) =>
   countryList.find((country) => country.value === countryCode);
 
-module.exports = { yearFrom, getCountry };
+function addIndividualFieldErrorMessages(validator, req, values) {
```
Comment (Caitlin-cooling) [inline src/lib/helpers.js:11]: Nice, like to see this stuff here

Comment (Caitlin-cooling) [reply src/lib/helpers.js:11]: Please could we have some tests?

**Review by mikebeeby — DISMISSED** [review]

File: src/locales/cy/default.yml [hunk 4]
```diff
@@ -97,7 +97,7 @@ validation:
   isUkPostcode: "Rhowch god post yn y DU"
   alphaNumericWithSpecialChars: "Ni ddylai eich {{ label }} gynnwys unrhyw nodau neu symbolau arbennig"
   default: "Mae'n rhaid i chi ateb y cwestiwn hwn"
-  houseNameOrHouseNumber: "Rhowch enw tŷ neu rif tŷ"
+  ukBuildingAddressEmptyValidator: "Rhowch rif y neu enw tŷ neu rhif fflat"
```
Comment (mikebeeby) [inline src/locales/cy/default.yml:100]: Hmm not sure this is right `Rhowch rif y neu enw tŷ neu rhif fflat` ins Google translate is `Enter the number or house name or flat number`

Comment (mikebeeby) [reply src/locales/cy/default.yml]: Changed to `Rhowch rif y neu enw tŷ`

**Pull request merged**